src/transformers/trainer.py

```diff
@@ -1956,6 +1978,17 @@ def _inner_training_loop(
         # Wait for the checkpoint to be uploaded.
         self._finish_current_push()
 
+        # After training we make sure to retrieve back the original forward pass method
+        # for the embedding layer by removing the forward post hook.
+        if self.neftune_noise_alpha is not None:
+            if is_peft_available() and isinstance(self.model, PeftModel):
+                embeddings = unwrap_model(self.model.base_model).get_input_embeddings()
+            else:
+                embeddings = unwrap_model(self.model).get_input_embeddings()
```

Comment:
Is this logic used anywhere else? It looks general enough that we could have a `_get_model_input_embeddings` function (not necessarily to be done in this PR)

Reply:
happy to refactor this in a follow up PR!